tests/pipeline/test_argument_inference.py

```diff
@@ -0,0 +1,247 @@
+        "argument": Argument(
+            name="argument",
+            type=str,
+            optional=False,
+        ),
+    }
+
+
+def test_types():
+    class TestComponent(PandasTransformComponent):
+        def __init__(
+            self,
+            *,
+            str_argument: str,
+            int_argument: int,
+            float_argument: float,
+            bool_argument: bool,
+            dict_argument: dict,
+            list_argument: list,
+            **kwargs,
+        ):
+            pass
+
+    assert infer_arguments(TestComponent) == {
+        "str_argument": Argument(
+            name="str_argument",
+            type=str,
+            optional=False,
+        ),
+        "int_argument": Argument(
+            name="int_argument",
+            type=int,
+            optional=False,
+        ),
+        "float_argument": Argument(
+            name="float_argument",
+            type=float,
+            optional=False,
+        ),
+        "bool_argument": Argument(
+            name="bool_argument",
+            type=bool,
+            optional=False,
+        ),
+        "dict_argument": Argument(
+            name="dict_argument",
+            type=dict,
+            optional=False,
+        ),
+        "list_argument": Argument(
+            name="list_argument",
+            type=list,
+            optional=False,
+        ),
+    }
+
+
+def test_optional_types():
+    class TestComponent(PandasTransformComponent):
+        def __init__(
+            self,
+            *,
+            str_argument: t.Optional[str] = "",
+            int_argument: t.Optional[int] = 1,
+            float_argument: t.Optional[float] = 1.0,
+            bool_argument: t.Optional[bool] = False,
+            dict_argument: t.Optional[dict] = None,
+            list_argument: t.Optional[list] = None,
+            **kwargs,
+        ):
+            pass
+
+    assert infer_arguments(TestComponent) == {
+        "str_argument": Argument(
+            name="str_argument",
+            type=str,
+            optional=True,
+            default="",
+        ),
+        "int_argument": Argument(
+            name="int_argument",
+            type=int,
+            optional=True,
+            default=1,
+        ),
+        "float_argument": Argument(
+            name="float_argument",
+            type=float,
+            optional=True,
+            default=1.0,
+        ),
+        "bool_argument": Argument(
+            name="bool_argument",
+            type=bool,
+            optional=True,
+            default=False,
+        ),
+        "dict_argument": Argument(
+            name="dict_argument",
+            type=dict,
+            optional=True,
+            default=None,
+        ),
+        "list_argument": Argument(
+            name="list_argument",
+            type=list,
+            optional=True,
+            default=None,
+        ),
+    }
+
+
+def test_parametrized_types_old():
```

Comment:
what's the "old" referring to? can we keep it to `test_parametrized_types` to match the above?